tar/bsdtar.c

```diff
@@ -231,6 +231,9 @@ main(int argc, char **argv)
 	/* Default: Perform basic security checks. */
 	bsdtar->extract_flags |= SECURITY;
 
+	/* Default: Extract atomically if possible */
+	bsdtar->extract_flags |= ARCHIVE_EXTRACT_ATOMIC;
+
```

Comment:
As already commented I don't want it to be default for now. We will need an self-explanatory option "--safe-writes" or something similar.